mat64/dense_arithmetic.go

```diff
@@ -351,6 +351,7 @@ func (m *Dense) Dot(b Matrix) float64 {
 }
 
 // Mul takes the matrix product of a and b, placing the result in the receiver.
+// If a or b are also the receiver, a workspace will be allocated.
```

Comment:
Can you explain why "workspace" is confusing? The reason I would like to include it is that it signifies a large allocation rather than just some memory being allocated. The term is used in other matrix code that is allocating (e.g csparse and sparseBLAS).